File: sohee-jung/week5/백준_silver_수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        int[] A = new int[N];

        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(A);

        int M = Integer.parseInt(br.readLine());
        int[] B = new int[M];

        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < M; i++) {
            B[i] = Integer.parseInt(st.nextToken());
        }

        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < M; i++) {
            if (binarySearch(A, B[i])) {
                sb.append(1).append('\n');
            } else {
                sb.append(0).append('\n');
            }
        }

        System.out.print(sb);
    }

    private static boolean binarySearch(int[] arr, int key) {
        int left = 0;
        int right = arr.length - 1;

        while (left <= right) {
            int mid = (left + right) / 2;

            if (arr[mid] == key) {
                return true;
            } else if (arr[mid] < key) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return false;
    }
}

File: minseo-song/week5/백준_silver_수 찾기.kt
package week5

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

fun main(){
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val n = bufferedReader.readLine().toInt()
    val a = bufferedReader.readLine().split(" ").map { it.toInt() }.sorted()
    val m = bufferedReader.readLine().toInt()
    val queries = bufferedReader.readLine().split(" ").map { it.toInt() }

    for (q in queries) {
        if (binarySearch(a, q)) {
            bufferedWriter.write("1\n")
        } else {
            bufferedWriter.write("0\n")
        }
    }

    bufferedWriter.close()
    bufferedReader.close()
}

fun binarySearch(arr: List<Int>, target: Int): Boolean{
    var left = 0
    var right = arr.size -1

    while (left<=right){
        val mid = (left+right)/2
        when{
            arr[mid] == target -> return true
            arr[mid] < target -> left = mid +1
            else -> right = mid -1
        }
    }

    return false
}